### PR TITLE
Remove Group and GroupSchema for TileDB-CF-Py

### PR DESCRIPTION
**Pull request by jp-dark**
These classes were used before the Group class was added to TileDB and TileDB-Py. There functionality can now be fully replaced with the standard TileDB objects.

### CODE REVIEW

File: tiledb/cf/netcdf_engine/converter.py [hunk 1]
```diff
@@ -795,19 +751,34 @@ def copy_to_array(
...
+                    tiledb_timestamp=timestamp,
+                    assigned_dim_values=assigned_dim_values,
+                    assigned_attr_values=assigned_attr_values,
+                    copy_metadata=copy_metadata,
+                )
+            if copy_metadata:
+                with tiledb.open(
+                    output_uri, mode="w", key=key, timestamp=timestamp, ctx=ctx
+                ) as array:
+                    copy_group_metadata(netcdf_group, array.meta)
```
Comment (perrygeo) [inline tiledb/cf/netcdf_engine/converter.py:781]: good call, no need for a private function here.

File: tiledb-cf-spec.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 :information_source: **Notes:**
 
-* The current TileDB-CF format version number is **0.2.0**.
```
Comment (perrygeo) [inline tiledb-cf-spec.md]: Some questions I might have from the user's perspective re: spec versions

- What version is my current data? We could record this in the metadata at write time but  we may want to retain a summary of the old spec to identify un-versioned data.
- There's a newer version of the spec. How do I upgrade/migrate my old spec to new? This implies we need an upgrade guide and/or migration scripts for each spec version jump.
- How does the spec version co-evolve with the software version re: backwards compatibility? Will every spec bump be a major version bump? We could include a sentence or two as to why we need two separate versions here.

**Review by perrygeo — APPROVED** [review]
Looks great. I believe all the old-style group logic have now been removed in favor of proper groups?

A few suggestions re: spec versioning but no blockers.

File: tiledb-cf-spec.md (hunk 2 above)
Comment (jp-dark) [reply tiledb-cf-spec.md]: These are all important questions. I'm going to roll-back the spec change and make a separate issue that needs to be addressed before the next release.

**Pull request merged**